src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfigValueSerializer`1.cs

```diff
@@ -21,6 +21,8 @@ internal sealed class EditorConfigValueSerializer<T> : IEditorConfigValueSeriali
         private readonly Func<string, Optional<T>> _parseValue;
         private readonly Func<T, string> _serializeValue;
 
+        private readonly ConcurrentDictionary<string, Optional<T>> _cachedValues = new();
```

Comment:
📝 Note that this type has copy-on-write behavior, so if there are a large number of _distinct_ strings for a single setting, it could degrade in performance.

Reply:
i can look into that.  after this change though, i saw no more allocations here.